quartz/components/scripts/search.inline.ts

```diff
@@ -190,22 +282,53 @@ document.addEventListener("nav", async (e: unknown) => {
             field: "content",
             tokenize: "reverse",
           },
+          {
+            field: "tags",
+            tokenize: "reverse",
+          },
         ],
       },
     })
 
-    let id = 0
-    for (const [slug, fileData] of Object.entries<ContentDetails>(data)) {
-      await index.addAsync(id, {
-        id,
-        slug: slug as FullSlug,
-        title: fileData.title,
-        content: fileData.content,
-      })
-      id++
-    }
+    fillDocument(index, data)
+  }
+
+  if (!tagIndex) {
+    // Initialize tag index (only indexes "tags" field but filled with same data)
+    tagIndex = new Document({
```

Comment:
is this as expensive as the first `fillDocument` call? I remember having some perf issues for some vaults >4000 files. I wonder how we can test this

Reply:
I don't have a big vault to test this with, currently using 4 small files for testing, which take ~`0.023ms` avg for creating document and ~`0.025ms` avg for filling the document. 

I saw that flexsearch has a `import` and `export` function for exporting all data. The document will still need to be created every time (and unfortunately cant be reused with slight adjustments so it could be used for both tags and regular from what I found), but that should work in static time. 

Ideally, the `export` could happen at build time (unless there's a way new routes/documents can be generated at runtime that im unaware of?). The result would need to be stored in `localStorage` or designated files in `.quartz-cache` (probably better solution since it has no size limit, unless it turns out reading it back in is too expensive).

Afterwards, in `search`, we would just need to create the document and read the pre-build indexed data back in.

This would be a separate pull request, but if the performance overhead for reading having flexsearch parse data from the file is too expensive for small vaults, we could use some heuristic to only start caching after a certain size.

Reply:
Hm I spent a bit more time looking at FlexSearch docs this morning, _seems_ like we can probably get away with a single index and then searching by field on that index. Could you take a look at that?

Reply:
cool, i was looking for something like that but didnt come across this, will check it out!

Reply:
done in 5a80dfc210675584e2817d82e7637e042a0bb515.

apperently, they renamed the property from `field` to `index`, but the shared index now works with basic search (still gets hits for content and title) + tags as before.